### PR TITLE
v2: Prepare module for v2 release

### PR DESCRIPTION
**Pull request by michaeldiamant**
Follows #449 to prepare go-algorand-sdk Go module for a v2 release by updating its module definition.  

* See https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher for context.
* Source file changes made mechanically via ` find . -type f -exec sed -i "" 's|"github.com/algorand/go-algorand-sdk/|"github.com/algorand/go-algorand-sdk/v2/|g' {} \;`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,6 +14,5 @@ require (
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
-	golang.org/x/sys v0.0.0-20211019181941-9d821ace8654 // indirect
```
Comment (algochoi) [inline go.mod:17]: Is this from running `go mod tidy`?

**Review by algochoi — APPROVED** [review]
Looks reasonable - example of bumping versions in the wild: https://github.com/uber-go/tally/blob/master/go.mod

File: go.mod (hunk 1 above)
Comment (michaeldiamant) [reply go.mod:17]: @algochoi Correct.

**Review by ahangsu — APPROVED** [review]
I assume we are taking **Major branch** option in https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher, rather than creating major subdirectory, then this PR should look good to me.

**Review by jasonpaulos — APPROVED** [review]
The changes match what I was expecting

**Pull request merged**